File: 767-prime-number-of-set-bits-in-binary-representation/prime-number-of-set-bits-in-binary-representation.py
class Solution:
    def checkPrime(self, num):
        if num <= 1:
            return 0
        
        for i in range(2, int(num**0.5)+1, 1):
            if num % i == 0:
                return 0
        return 1

    def countPrimeSetBits(self, left: int, right: int) -> int:
        ans = 0

        for i in range(left, right+1):
            setBits = bin(i).count('1')
            if (self.checkPrime(setBits)):
                ans += 1
        
        return ans

File: 767-prime-number-of-set-bits-in-binary-representation/prime-number-of-set-bits-in-binary-representation.cpp
class Solution {
public:
    int checkPrime(int num) {
        if (num == 1 || num == 0)
            return 0;

        for (int i=2; i*i <= num; i++) {
            if (num % i == 0)
                return 0;
        }
        return 1;
    }
    int countPrimeSetBits(int left, int right) {
        int ans = 0;
        for (int i=left; i<=right; i++) {
            // C++ useful function depend on compiler -> O(1)
            int bits = __builtin_popcount(i);
            ans += checkPrime(bits);
        }
        return ans;
    }
};